client/my-sites/upgrades/cart/test/cart-buttons.js

```diff
@@ -1,65 +1,68 @@
 /**
  * External Dependencies
  */
-const sinon = require( 'sinon' ),
-	ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	TestUtils = require( 'react-addons-test-utils' ),
-	{ expect } = require( 'chai' );
+import sinon from 'sinon';
+import React from 'react';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import identity from 'lodash/identity';
 
 /**
  * Internal dependencies
  */
 import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
 
 describe( 'cart-buttons', function() {
-	useFakeDom();
-
-	beforeEach( function() {
-		this.CartButtons = require( '../cart-buttons.jsx' );
-		this.CartButtons.prototype.translate = sinon.stub();
-		this.recordStub = this.CartButtons.prototype.__reactAutoBindMap.recordEvent = sinon.stub();
+	const recordStub = sinon.stub();
+	const AnalyticsMixinStub = () => ( {
+		recordEvent: recordStub
 	} );
+	let CartButtons;
+
+	useFakeDom();
 
-	afterEach( function() {
-		delete this.CartButtons.prototype.translate;
-		delete this.CartButtons.prototype.__reactAutoBindMap.recondEvent;
+	useMockery( mockery => {
+		mockery.registerMock( 'lib/mixins/analytics', AnalyticsMixinStub );
+		CartButtons = require( '../cart-buttons.jsx' ).CartButtons;
 	} );
 
 	describe( 'Click on Keep Searching Button', function() {
 		beforeEach( function() {
 			this.onKeepSearchingClick = sinon.stub();
-			this.cartButtonsComponent = TestUtils.renderIntoDocument(
-				<this.CartButtons
+			this.cartButtonsComponent = mount(
+				<CartButtons
 					selectedSite={ {slug: 'example.com'} }
 					showKeepSearching={ true }
 					onKeepSearchingClick={ this.onKeepSearchingClick }
+					translate={ identity }
 					/>
 			);
 		} );
 
 		it( 'should track "keepSearchButtonClick" event', function() {
-			TestUtils.Simulate.click( ReactDom.findDOMNode( this.cartButtonsComponent.refs.keepSearchingButton ) );
-			expect( this.recordStub ).to.have.been.calledWith( 'keepSearchButtonClick' );
+			this.cartButtonsComponent.find( '.cart-keep-searching-button' ).simulate( 'click' );
+			expect( recordStub ).to.have.been.calledWith( 'keepSearchButtonClick' );
```

Comment:
I would really suggest using the sandbox helper whenever we use sinon, because we are using a single test runner. It's easy to miss stub cleanup otherwise.